File: benchmark/agbenchmark/agent_protocol_client/models/step_request_body.py
import json
from dataclasses import dataclass
from typing import Any, Optional, Union

@dataclass
class StepRequestBody:
    """
    Body of the task request.
    """

    input: Optional[str] = None
    additional_input: Optional[Union[str, int, float, bool, dict, list]] = None

    def to_json(self) -> str:
        """Returns the JSON representation of the model"""
        return json.dumps(self, default=str)

File: frontend/lib/models/step_request_body.dart
class StepRequestBody {
  final String? input;
  final Map<String, dynamic>? additionalInput;

  StepRequestBody({required this.input, this.additionalInput});

  Map<String, dynamic> toJson() {
    final data = <String, dynamic>{};
    if (input != null) {
      data['input'] = input;
    }
    if (additionalInput != null) {
      data['additional_input'] = additionalInput;
    }
    return data;
  }

  /// Returns `true` if both `input` and `additionalInput` are `null`.
  bool get isEmpty => input == null && additionalInput == null;
}

